cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -100,10 +103,13 @@ __launch_bounds__(int(ChainedPolicyT::ActivePolicy::AgentLargeBufferPolicyT::BLO
   using StatusWord    = typename TileT::StatusWord;
   using ActivePolicyT = typename ChainedPolicyT::ActivePolicy::AgentLargeBufferPolicyT;
   using BufferSizeT   = cub::detail::value_t<BufferSizeIteratorT>;
+  /// Internal load/store type. For byte-wise memcpy, a single-byte type
+  using AliasT = typename ::cuda::std::
+    conditional<IsMemcpy, char, cub::detail::value_t<cub::detail::value_t<InputBufferIt>>>::type;
```

Comment:
We need to avoid the template instantiation of `iterator_traits` (via `cub::detail::value_t`) by deferring the `iterator_traits::type` member type access to avoid the compilation error for iterator-to-pointer-to-void types (`error: reference to void is not allowed`).
```suggestion
using AliasT = typename ::cuda::std::conditional<
    IsMemcpy,
    std::iterator_traits<char *>,
    std::iterator_traits<cub::detail::value_t<InputBufferIt>>>::type::value_type;
```